### PR TITLE
remove logic for archive table migration

### PR DESCRIPTION
**Pull request by tomrf1**
the migration is complete.

Also, `DynamoArchivedChannelTests` and `DynamoChannelTests` now have some duplicated logic that will not be going away. So I've moved this logic into a new `DynamoService` abstract class, which they both inherit from.

### CODE REVIEW

File: app/services/DynamoArchivedChannelTests.scala [hunk 1]
```diff
@@ -2,65 +2,13 @@ package services
...
-import models._
 import software.amazon.awssdk.services.dynamodb.DynamoDbClient
 import software.amazon.awssdk.services.dynamodb.model._
-import zio.blocking.effectBlocking
-import zio.duration.durationInt
-import zio.stream.ZStream
 import zio.{ZEnv, ZIO}
 
-import scala.jdk.CollectionConverters._
+class DynamoArchivedChannelTests(stage: String, client: DynamoDbClient) extends DynamoService(stage, client) with StrictLogging {
```
Comment (tomrf1) [inline app/services/DynamoArchivedChannelTests.scala:9]: The new `DynamoService` class now provides most of the logic for this file

File: app/services/DynamoChannelTests.scala [hunk 2]
```diff
@@ -77,26 +77,6 @@ class DynamoChannelTests(stage: String, client: DynamoDbClient) extends StrictLo
       ).items
     }.mapError(DynamoGetError)
 
-  // TODO - migrate to use the new archived tests table
-  def getAllArchived(channel: Channel): ZIO[ZEnv, DynamoGetError, java.util.List[java.util.Map[String, AttributeValue]]] =
```
Comment (tomrf1) [inline app/services/DynamoChannelTests.scala:81]: this function was only used for the migration

File: app/services/DynamoChannelTests.scala [hunk 3]
```diff
@@ -112,26 +92,6 @@ class DynamoChannelTests(stage: String, client: DynamoDbClient) extends StrictLo
       ).items
     }.mapError(DynamoGetError)
 
-  // Sends a batch of write requests, and returns any unprocessed items
-  private def putAll(writeRequests: List[WriteRequest]): ZIO[ZEnv, DynamoPutError, List[WriteRequest]] =
```
Comment (tomrf1) [inline app/services/DynamoChannelTests.scala:116]: now provided by `DynamoService`

File: app/services/DynamoService.scala [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import software.amazon.awssdk.services.dynamodb.model.{BatchWriteItemRequest, ReturnConsumedCapacity, WriteRequest}
+import zio.{ZEnv, ZIO}
+import zio.blocking.effectBlocking
+import zio.stream.ZStream
+import zio.duration.durationInt
+
+import scala.jdk.CollectionConverters._
+
+// Shared functionality for DynamoDb services
+abstract class DynamoService(stage: String, client: DynamoDbClient) extends StrictLogging {
```
Comment (tomrf1) [inline app/services/DynamoService.scala:15]: Nothing here is new - it's copied from the existing dynamo classes

**Review by KaliedaRik — APPROVED** [review]
LGTM. Next job is to fix the RRCP campaigns page?

**Pull request merged**